src/components/PrioritySelector/PrioritySelector.tsx

```diff
@@ -0,0 +1,147 @@
+    onSelect: (newPriorityItem: PriorityItemType<PriorityItemValue>) => void;
+    /** Boolean indicating whether the whole badge should be disabled. */
+    isDisabled?: boolean;
+    /** Priority button label name for ARIA labelling */
+    buttonLabel?: string;
+    /** ref to possible parent div, if it is a badge */
+    badgeRef?: React.RefObject<HTMLElement | null> | React.ForwardedRef<HTMLDivElement>;
+    /** ref to iconButton, used for keyboard navigation */
+    buttonRef?: React.RefObject<HTMLElement>;
+}
+
+const { block, elem } = bem('PrioritySelector', styles);
+const itemStylesBem = bem('DropdownItem', itemStyles);
+export function PrioritySelector<PriorityItemValue>({
+    onSelect,
+    selectedItem,
+    isDisabled = false,
+    buttonLabel,
+    list,
+    badgeRef,
+    buttonRef,
+    ...rest
+}: Props<PriorityItemValue>) {
+    const renderPriorityIcon = (priorityType?: Priority, disabled: boolean = false) => {
+        if (!priorityType) {
+            return <></>;
+        }
+        const IconComponent = iconMap[priorityType];
+        return IconComponent ? (
+            <IconComponent
+                {...elem('icon', { [priorityType]: true })}
+                disabled={disabled}
+                viewBox="0 0 24 24"
+            />
+        ) : (
+            <></>
+        );
+    };
+
+    const priorityOrder: Record<Priority, number> = {
+        [Priority.Mandatory]: 1,
+        [Priority.Important]: 2,
+        [Priority.Optional]: 3,
+        [Priority.Exclude]: 4,
+    };
+
+    // Sort the list based on the priority order
+    const sortedList = list
+        .slice()
+        .sort((a, b) => priorityOrder[a.priority] - priorityOrder[b.priority]);
+
+    return (
+        <Root {...block({ ...rest })} refElement={badgeRef}>
+            <DropdownTrigger>
+                <IconButton
+                    {...elem('icon', { [selectedItem.priority]: true })}
+                    variant="ghost"
+                    aria-label={buttonLabel || selectedItem.label}
+                    disabled={isDisabled}
+                    type="button"
+                    ref={buttonRef}
+                >
+                    {renderPriorityIcon(selectedItem.priority, isDisabled)}
+                </IconButton>
+            </DropdownTrigger>
+            <Portal>
+                <DropdownContent
+                    {...elem('badgeDropdownList', { fixedWidth: !badgeRef })}
+                    sideOffset={6}
+                    refElement={badgeRef}
+                >
+                    {sortedList.map((item) => (
+                        <DropdownMenuItem
+                            key={item.priority}
+                            onSelect={(e) => {
+                                e.stopPropagation();
+                                onSelect(item);
+                            }}
+                            role="option"
+                            aria-selected={selectedItem === item}
```

Comment:
I wonder if this can go wrong with JS object comparison, e.g. if the objects have the same values, but created separately. Maybe just to be sure, it would be better to compare a string value, like `selectedItem.priority === item.priority`?